### PR TITLE
Fix downsampling bug if non-default file_len is used

### PR DESCRIPTION
**Pull request by rschwanhold**
fixes: #133

### CODE REVIEW

File: wkcuber/downsampling.py [hunk 1]
```diff
@@ -148,15 +148,15 @@ def downsample(
     logging.debug(
         "Found source cubes: count={} size={} min={} max={}".format(
             len(source_cube_addresses),
-            (cube_edge_len,) * 3,
+            (buffer_edge_len,) * 3,
```
Comment (rschwanhold) [inline wkcuber/downsampling.py]: If no explicit `buffer_edge_len` is specified, the value is `None` at this point. This leads to debug messages like: `DEBUG Found source cubes: count=1 size=(None, None, None) min=(0, 0, 0) max=(0, 0, 0)`. 
How should we handle this?

File: wkcuber/downsampling.py [hunk 2]
```diff
@@ -168,7 +168,10 @@ def downsample(
             wkw.Header.BLOCK_TYPE_LZ4HC if compress else wkw.Header.BLOCK_TYPE_RAW
         )
         ensure_wkw(
-            target_wkw_info, block_type=header_block_type, num_channels=num_channels
+            target_wkw_info,
```
Comment (philippotto) [inline wkcuber/downsampling.py]: Nice stuff! I created the follow up issue we talked about: https://github.com/scalableminds/webknossos-cuber/issues/142

File: wkcuber/downsampling.py (hunk 1 above)
Comment (philippotto) [reply wkcuber/downsampling.py]: I suggest to extract https://github.com/scalableminds/webknossos-cuber/pull/141/files#diff-363a076e73484d6f38615890702e0506R224-R226 into a `determine_buffer_edge_len` method and then open the source dataset before logging these values. Then, you can read the file_len and print the buffer_edge_len afterwards properly.

**Comment by philippotto** [comment]
@normanrz the license/cla check doesn't finish. Can you have a look?

**Pull request merged**